src/services/GenerateVerifiableCredential.ts

```diff
@@ -0,0 +1,457 @@
+	*/
+  private calculateVerificationProcessLevel(faceMatchCheck: string): number {
+  	return faceMatchCheck === YotiSessionDocument.APPROVE ? 3 : 0;
+  }
+	
+  private getCounterIndicator(
+  	ID_DOCUMENT_FACE_MATCH_RECOMMENDATION: any,
+  	ID_DOCUMENT_AUTHENTICITY_RECOMMENDATION: any,
+  	IBV_VISUAL_REVIEW_CHECK_RECOMMENDATION?: any,
+  	DOCUMENT_SCHEME_VALIDITY_CHECK_RECOMMENDATION?: any,
+  	PROFILE_DOCUMENT_MATCH_RECOMMENDATION?: any,
+  ): any {
+  	const ci: string[] = [];
+	
+  	const addToCI = (code: string | string[]) => {
+  		if (Array.isArray(code)) {
+  			ci.push(...code);
+  		} else {
+  			ci.push(code);
+  		}
+  	};
+	
+  	const handleFaceMatchRejection = () => {
+  		const { value, reason } = ID_DOCUMENT_FACE_MATCH_RECOMMENDATION;
+  		if (value === "REJECT") {
+  			switch (reason) {
+  				case "FACE_NOT_GENUINE":
+  				case "LARGE_AGE_GAP":
+  				case "PHOTO_OF_MASK":
+  				case "PHOTO_OF_PHOTO":
+  				case "DIFFERENT_PERSON":
+  					addToCI("V01");
+  					break;
+  				default:
+  					break;
+  			}
+  		}
+  	};
+	
+  	const handleAuthenticityRejection = () => {
+  		const { value, reason } = ID_DOCUMENT_AUTHENTICITY_RECOMMENDATION;
+  		if (value === "REJECT") {
+  			switch (reason) {
+  				case "COUNTERFEIT":
+  					addToCI("D14");
+  					break;
+  				case "EXPIRED_DOCUMENT":
+  					addToCI("D16");
+  					break;
+  				case "FRAUD_LIST_MATCH":
+  					addToCI(["F03", "D14"]);
+  					break;
+  				case "DOC_NUMBER_INVALID":
+  					addToCI("D02");
+  					break;
+  				case "TAMPERED":
+  				case "DATA_MISMATCH":
+  				case "CHIP_DATA_INTEGRITY_FAILED":
+  				case "CHIP_SIGNATURE_VERIFICATION_FAILED":
+  				case "CHIP_CSCA_VERIFICATION_FAILED":
+  					addToCI("D14");
+  					break;
+  				default:
+  					break;
+  			}
+  		}
+  	};
+	
+  	handleFaceMatchRejection();
+  	handleAuthenticityRejection();
+	
+  	return ci;
+  }
+
+  private attachPersonName(
+  	credentialSubject: VerifiedCredentialSubject,
+  	givenName: string,
+  	familyName: string,
+  ): VerifiedCredentialSubject {
+  	const givenNames = givenName.split(" ");
+  	const nameParts = givenNames.map((name) => ({ value: name, type: "GivenName" }));
+	
+  	nameParts.push({ value: familyName, type: "FamilyName" });
+	
+  	credentialSubject.name = [{ nameParts }];
+	
+  	return credentialSubject;
+  }
+
+  private attachDOB(
+  	credentialSubject: VerifiedCredentialSubject,
+  	dateOfBirth: string,
+  ): VerifiedCredentialSubject {
+  	credentialSubject.birthDate = [{ value: dateOfBirth }];
+	
+  	return credentialSubject;
+  }
+	
+
+  private attachAddressInfo(
+  	credentialSubject: VerifiedCredentialSubject,
+  	postalAddress: any,
+  ): VerifiedCredentialSubject {
+  	credentialSubject.address = [
+  		{
+  			buildingNumber: postalAddress.building_number,
+  			addressLocality: postalAddress.town_city,
+  			postalCode: postalAddress.postal_code,
+  			addressCountry: postalAddress.country,
```

Comment:
did we get agreement on the full address parts mapping? this misses the street address for instance

Reply:
streetName has been added, other fields Anand to confirm at future date if required